src/us/kbase/workspace/database/provenance/Title.java

```diff
@@ -0,0 +1,171 @@
+package us.kbase.workspace.database.provenance;
+
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.Optional;
+import us.kbase.workspace.database.Util;
+
+/**
+ * Represents a title or name of a resource.
+ */
+public class Title {
+
+	private final String titleString;
+	private final TitleType titleType;
+	private final String titleLanguage;
+
+	/**
+	 * An enum representing types of title.
+	 */
+	public enum TitleType {
+		SUBTITLE,
```

Comment:
Shouldn't just TITLE be one of the options? Or does the lack of a title type imply it's a regular title?

Reply:
yes -- titleType is only used if it's not a plain old title.

Reply:
Can we document that? I don't think the documentation makes that clear

Reply:
Alternatively, add TITLE and require the enum... that might be simpler and more explicit

Reply:
And potentially add UNKNOWN as a default if there's no info

Reply:
The info about the "missing" default is in the docs for the schema but not here, so that can easily be corrected. I'll do the same with the assumed language.

Reply:
Update: added the default `TITLE` value to the enum.

Reply:
:+1: